DOC-2481: 7.3 Revision History changes - update plugin page and add release notes (#3407)
tinymce/tinymce-docs: modules/ROOT/pages/revisionhistory.adoc

== Data structure

=== Revision

The revision is an `+Object+` that contains the following fields:

[cols="1,1,1,3",options="header"]
|===
| Field          | Type       | Required? | Description
| `+revisionId+` | `+string+` | required  | The unique string ID of the revision.
| `+createdAt+`  | `+string+` | required  | A UTC datetime string in ISO-8061 format.
| `+content+`    | `+string+` | optional  | HTML string of the revision content. Empty string is considered as valid content.
| `+author+`     | xref:#author[Author] `+Object+` | optional  | The author of the revision.
|===

=== Author

The author is an `+Object+` that represents the author or creator of a revision. It contains the following fields:

[cols="1,1,1,3",options="header"]
|===
| Field      | Type       | Required? | Description
| `+id+`     | `+string+` | required  | The unique string ID of the author.
| `+name+`   | `+string+` | optional  | The name of the revision author. If not provided, the default value is `Anonymous`.

> | `+avatar+` | `+string+` | optional  | The URL of the author's avatar image. If not provided, the {pluginname} will use a generated avatar using the author's initials.

Reviewer: ```suggestion
| `+avatar+` | `+string+` | optional  | The URL for the author's avatar image. If not provided or invalid, the Revision History will use a generated avatar featuring the author's initials.
```

> | `+avatar+` | `+string+` | optional  | The URL of the author's avatar image. If not provided, the {pluginname} will use a generated avatar using the author's initials.

Reviewer: ```suggestion
| `+avatar+` | `+string+` | optional  | The URL of the author's avatar image. If not provided or invalid, the {pluginname} will use a generated avatar using the author's initials.
```

  })
});
----

== Understanding types of revision

There are three types of revision in {pluginname} plugin:

* `Initial`: This revision is generated during the {productname} `Loaded` event, with the editor's initial content.
* `Draft`: It is generated upon opening the {pluginname}, reflecting the editor's current content. When included, it becomes the `latest revision` in the revisions list.
* `Saved`: These revisions are fetched from the client's storage upon opening the {pluginname} view, using the `revisionhistory_fetch` and `revisionhistory_fetch_revision` options.

When dealing with a document that has no saved revisions, the {pluginname} plugin will typically maintain two revisions: `initial` and `draft`. If there are no changes to the content of the editor since initialization, the `initial` revision is not displayed since it's identical to the `draft` revision.

For documents with saved revisions, the `initial` revision is disregarded, assuming it already exists in the saved revisions. If the integrators want to include the initial content, a workaround is to include it as a revision via `revisionhistory_fetch` option.

== Data structure

=== Revision

> The revision an is `+Object+` that contains the following fields:

Author: ```suggestion

The revision is an `+Object+` that contains the following fields:
```
